x-pack/plugins/security/public/views/management/roles_grid/components/confirm_delete/confirm_delete.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import React, { Component, Fragment } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { RolesApi } from '../../../../../lib/roles_api';
+
+interface Props {
+  rolesToDelete: string[];
+  intl: InjectedIntl;
+  callback: (rolesToDelete: string[], errors: string[]) => void;
+  onCancel: () => void;
+}
+
+class ConfirmDeleteUI extends Component<Props, {}> {
+  public deleteUsers = () => {
+    const { rolesToDelete, callback } = this.props;
+    const errors: string[] = [];
+    rolesToDelete.forEach(async roleName => {
+      try {
+        await RolesApi.deleteRole(roleName);
+        toastNotifications.addSuccess(
+          this.props.intl.formatMessage(
+            {
+              id:
+                'xpack.security.management.roles.confirmDelete.roleSuccessfullyDeletedNotificationMessage',
+              defaultMessage: 'Deleted role {roleName}',
+            },
+            { roleName }
+          )
+        );
+      } catch (e) {
+        errors.push(roleName);
+        toastNotifications.addDanger(
+          this.props.intl.formatMessage(
+            {
+              id:
+                'xpack.security.management.roles.confirmDelete.roleDeletingErrorNotificationMessage',
+              defaultMessage: 'Error deleting role {roleName}',
+            },
+            { roleName }
+          )
+        );
+      }
+
+      callback(rolesToDelete, errors);
```

Comment:
We're calling the `callback` for every role which we delete before they're all deleted.

Reply:
oh yikes...turns out this was being used for two purposes, one of which required an immediate call. I need to rework this to behave like the confirmation modal we have when deleting spaces.